src/registrar/views/application.py

```diff
@@ -329,6 +335,41 @@ def pending_applications(self):
         ]
         return DomainApplication.objects.filter(creator=self.request.user, status__in=check_statuses)
 
+    def db_check_for_unlocking_steps(self):
+        """Helper for get_context_data
+
+        Queries the DB for an application and returns a dict for unlocked steps."""
+        history_dict = {
+            "organization_type": bool(self.application.organization_type),
+            "tribal_government": bool(self.application.tribe_name),
+            "organization_federal": bool(self.application.federal_type),
+            "organization_election": bool(self.application.is_election_board),
+            "organization_contact": (
+                bool(self.application.federal_agency)
+                or bool(self.application.organization_name)
+                or bool(self.application.address_line1)
+                or bool(self.application.city)
+                or bool(self.application.state_territory)
+                or bool(self.application.zipcode)
+                or bool(self.application.urbanization)
+            ),
+            "about_your_organization": bool(self.application.about_your_organization),
+            "authorizing_official": bool(self.application.authorizing_official),
+            "current_sites": (
+                bool(self.application.current_websites.exists()) or bool(self.application.requested_domain)
+            ),
+            "dotgov_domain": bool(self.application.requested_domain),
+            "purpose": bool(self.application.purpose),
+            "your_contact": bool(self.application.submitter),
+            "other_contacts": (
+                bool(self.application.other_contacts.exists()) or bool(self.application.no_other_contacts_rationale)
+            ),
+            "anything_else": (bool(self.application.anything_else) or bool(self.application.is_policy_acknowledged)),
+            "requirements": bool(self.application.is_policy_acknowledged),
+            "review": bool(self.application.is_policy_acknowledged),
+        }
+        return [key for key, value in history_dict.items() if value]
+
```

Comment:
bool() might run into issues with "falsy" data.  (eg. bool("") = false, bool(0) = false).  This might not be an issue, but stronger (more specific) data checks might be better.  I'd ask a second opinion.

Reply:
I could definitely see bool(0) being an issue if an int got passed in somehow. That said though, I don't see that happening in normal application flow.

It certainly doesn't hurt to make the logic here more rigorous against it, though

Reply:
Fixed, thank you both for catching this